### PR TITLE
Upgrade ApolloServer to version 4.7.0 in code and documentation

### PR DESCRIPTION
**Pull request by mjfwebb**
# Description

This PR updates our codebase to use the latest version of ApolloServer, version 4. Migrating to version 4 means that we no longer have a full batteries-included version of Apollo in a single import, so we also include the `@apollo/server/standalone` in several examples.

As an aside when testing some documentation code, I noted that the `Neo4jDatabaseInfo` class was not exported, so that has been included as a change in this PR as it related to the ApolloServer upgrade.

In our test code, we use express middleware as described in the [Apollo documentation](https://www.apollographql.com/docs/apollo-server/migration#migrate-from-apollo-server-express).

While updating the example code I have converted require statements to import statements and added missing imports to make individual examples more usable. Changing to import statements from require is something we will want to do for the rest of the documentation, but is out of scope for this particular PR.

I started to use the `#graphql` annotation to enable syntax highlighting instead of importing `gql`. This is also something we will want to do with the rest of the documentation.

Finally, we directly use `gql` from the `graphql-tag` package as this is no longer re-exported by ApolloServer.

## Complexity

Medium

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-3239.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 37 | 58 | 1134 |
|  aggregations.NestedAggregation | 14551 | 14551 | 75 | 114 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 59 | 69 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3423 | 3042 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3003 | 3127 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 45 | 52 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 57 | 66 | 2174 |
|  aggregations.DeeplyNestedCount | 13070329 | 13070329 | 4949 | 5286 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 133 | 164 | 600 |
|  connect.createAndConnect | 14424 | 14424 | 228 | 246 | 3003 |
|  connections.Connection | 14082 | 14082 | 60 | 87 | 2174 |
|  connections.NestedConnection | 41462 | 41462 | 163 | 177 | 4516 |
|  connections.ConnectionWithSort | 2282 | 2282 | 68 | 85 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2282 | 2282 | 216 | 353 | 1040 |
|  create.SimpleMutation | 7 | 7 | 59 | 71 | 1 |
|  cypher-directive.TopLevelCypherDirective | 116092 | 116092 | 619 | 760 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 147609 | 147609 | 183 | 203 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 243 | 326 | 1040 |
|  delete.NestedDeleteInUpdate | 22200 | 22200 | 158 | 199 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 62 | 50 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 95 | 122 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 3450 | 3450 | 32 | 35 | 1134 |
|  2925.NestedSingleRelationshipRequiredFilter | 9383 | 9383 | 53 | 61 | 1040 |
|  query.SimpleQuery | 15160 | 15160 | 47 | 53 | 2174 |
|  query.QueryWhere | 9692 | 9692 | 59 | 46 | 2161 |
|  query.SimpleQueryWithNestedWhere | 9870 | 9870 | 61 | 63 | 2161 |
|  query.Nested | 10092037 | 10092037 | 9984 | 10435 | 2008534 |
|  query.NestedWithFilter | 40401 | 40401 | 174 | 131 | 4511 |
|  query.OrFilterOnRelationships | 41841 | 42584 | 204 | 250 | 1361 |
|  query.OrFilterOnRelationshipsAndNested | 35546 | 36307 | 285 | 256 | 1361 |
|  query.QueryWithNestedIn | 13221 | 13197 | 108 | 67 | 1156 |
|  query.NestedConnectionWhere | 9834 | 9834 | 65 | 75 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9909 | 9898 | 98 | 105 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19140 | 19104 | 146 | 176 | 1598 |
|  query.NestedWithRelationshipSingleFilters | 3732 | 3732 | 185 | 283 | 1134 |
|  query.Fulltext | 80 | 80 | 34 | 39 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 46 | 65 | 84 |
|  sorting.SortMultipleTypes | 2513 | 2513 | 68 | 83 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1492 | 1456 | 208 | 304 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 54 | 57 | 2174 |
|  sorting.SortDeeplyNestedFields | 42196 | 42196 | 99 | 107 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 155 | 243 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 72 | 89 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 79 | 71 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 373 | 474 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 367 | 302 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 67 | 80 | 1040 |

</details>
        
Old Schema Generation: 44.059s
Schema Generation: 45.011s
Old Subgraph Schema Generation: 59.644s
Subgraph Schema Generation: 1:00.494 (m:ss.mmm)

File: docs/modules/ROOT/pages/getting-started.adoc [hunk 1]
```diff
@@ -26,7 +26,7 @@ cd neo4j-graphql-example
 +
 [source, bash, indent=0]
 ----
-npm init --yes
+npm init es6 --yes
```
Comment (tbwiss) [inline docs/modules/ROOT/pages/getting-started.adoc:29]: Shall we add a NOTE just underneath this code snippet to point out the usage of `es6`? Wdyt?

File: docs/modules/ROOT/pages/guides/apollo-federation.adoc [hunk 2]
```diff
@@ -61,20 +60,21 @@ const driver = neo4j.driver(
...
-    schema,
-  });
+const schema = neoSchema.getSubgraphSchema();
 
-  startStandaloneServer(server, { listen: { port: 4001 } }).then(({ url }) => {
-    console.log(`🚀  Server ready at ${url}`);
-  });
+const server = new ApolloServer({
+  schema,
+  resolvers,
```
Comment (tbwiss) [inline docs/modules/ROOT/pages/guides/apollo-federation.adoc:72]: I probably missed it but where are the `resolvers` defined?

**Review by tbwiss — COMMENTED** [review]
Thanks for this @mjfwebb!
Super nice commits in this PR, made it so much easier to review it!

File: docs/modules/ROOT/pages/guides/apollo-federation.adoc (hunk 2 above)
Comment (mjfwebb) [reply docs/modules/ROOT/pages/guides/apollo-federation.adoc:72]: Great spot! That was a mistake, it seems :+1:

File: docs/modules/ROOT/pages/getting-started.adoc (hunk 1 above)
Comment (mjfwebb) [reply docs/modules/ROOT/pages/getting-started.adoc:29]: Addressed in https://github.com/neo4j/graphql/pull/3239/commits/e0957d53f0af9d15077079d46601473a7dc52889

File: packages/graphql/tests/e2e/setup/apollo-server.ts [hunk 3]
```diff
@@ -92,16 +94,24 @@ export class ApolloTestServer implements TestGraphQLServer {
...
+            cors(),
+            bodyParser.json(),
+            expressMiddleware(server, {
+                // eslint-disable-next-line @typescript-eslint/require-await
+                context: async ({ req }) => ({ req }),
+            })
+        );
+
+        return new Promise<void>((resolve) =>
+            httpServer.listen({ port: 0 }, () => {
```
[the remark this answers is not in the document]
Comment (mjfwebb) [reply packages/graphql/tests/e2e/setup/apollo-server.ts]: Addressed in https://github.com/neo4j/graphql/pull/3239/commits/c817e3c096e1fcded3e25250c4b7751e0dfab895

File: docs/modules/ROOT/pages/guides/apollo-federation.adoc (hunk 2 above)
Comment (mjfwebb) [reply docs/modules/ROOT/pages/guides/apollo-federation.adoc]: Addressed in https://github.com/neo4j/graphql/pull/3239/commits/b917a0b4183a85b734d4750a00349dbaf9326a7c

File: docs/modules/ROOT/pages/guides/apollo-federation.adoc [hunk 4]
```diff
@@ -120,20 +119,21 @@ const driver = neo4j.driver(
...
-    schema,
-  });
+const schema = neoSchema.getSubgraphSchema();
 
-  startStandaloneServer(server, { listen: { port: 4002 } }).then(({ url }) => {
-    console.log(`🚀  Server ready at ${url}`);
-  });
+const server = new ApolloServer({
+  schema,
+  resolvers,
```
Comment (tbwiss) [inline docs/modules/ROOT/pages/guides/apollo-federation.adoc:130]: I think there is another `resolvers` in the `docs/modules/ROOT/pages/guides/apollo-federation.adoc` file that can be removed.

Comment (mjfwebb) [reply docs/modules/ROOT/pages/guides/apollo-federation.adoc]: https://github.com/neo4j/graphql/pull/3239/commits/6fafd2035b4b34651381c93b2d33d4513a2f6fe1 thanks!

**Review by tbwiss — APPROVED** [review]
LGTM now!

**Pull request merged**